### PR TITLE
[AC-2791] Members page - finish component library refactors

### PR DESCRIPTION
**Pull request by eliykat**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/AC-2791
## 📔 Objective

Follow-up refactors to make full use of the Component Library after #9515.

The main change is to make TableDataSource responsible for filtering, tracking, and updating its data. The motivation here is better encapsulation and to minimize logic that needs to be in the base class. This new functionality is contained in the `PeopleTableDataSource` subclass.
* move the status filtering logic into the `TableDataSource`'s filtering function
* move the user status count out of `statusMap` and into `TableDataSource`
* move a couple of utility methods (removing and updating users)
* move select/deselect logic

Other minor changes:
* delete unneeded `@ViewChild` properties and corresponding templating
* use ToastService for all toasts
* move revoke and restore methods from the base class into the subclass, as these are not actually shared between components (providerUsers cannot be revoked)

The next step is to update the provider users component to also use this base class and table data source.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/admin-console/organizations/members/people.component.ts [hunk 1]
```diff
@@ -63,27 +63,22 @@ import {
...
 
+class MemberTableDataSource extends PeopleTableDataSource<OrganizationUserView> {
+  protected statusType = OrganizationUserStatusType;
+}
+
 @Component({
   selector: "app-org-people",
   templateUrl: "people.component.html",
 })
 export class PeopleComponent extends NewBasePeopleComponent<OrganizationUserView> {
```
Comment (vincentsalucci) [inline apps/web/src/app/admin-console/organizations/members/people.component.ts:74]: 💭 I'm wondering if its finally time for us to align the name of this component all of the external references; `MembersComponent`

File: apps/web/src/app/admin-console/organizations/members/people.component.ts [hunk 2]
```diff
@@ -63,27 +63,22 @@ import {
 } from "./components/member-dialog";
 import { ResetPasswordComponent } from "./components/reset-password.component";
 
+class MemberTableDataSource extends PeopleTableDataSource<OrganizationUserView> {
```
Comment (vincentsalucci) [inline apps/web/src/app/admin-console/organizations/members/people.component.ts:66]: ⛏️ Completely non-blocking - lets update `MembersTableDataSource` to align with the plural use of the base class (Members == People)

**Review by vincentsalucci — COMMENTED** [review]
Overall this is **super** great! Thanks for making those preliminary talks into a reality. Just a couple of thoughts to discuss/move-on from. Excited to let QA vet these changes.

File: apps/web/src/app/admin-console/organizations/members/people.component.ts (hunk 1 above)
Comment (eliykat) [reply apps/web/src/app/admin-console/organizations/members/people.component.ts:74]: Good call! I've done that, however I've left the base class as `NewPeopleComponent` as the most general term between org members and provider users.

**Review by vincentsalucci — APPROVED** [review]

**Pull request merged**